### ISSUE TITLE
IListeningPort not provided by twisted.internet.udp.Port

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4462]
|[<img alt="ghazel's avatar" src="https://avatars.githubusercontent.com/u/31402?s=50" width="50" height="50">](https://github.com/ghazel)| @ghazel reported|
|-|-|
|Trac ID|trac#4462|
|Type|defect|
|Created|2010-05-29 00:35:07Z|
|Branch|https://github.com/twisted/twisted/tree/udp-listening-port-4462|

From http://twistedmatrix.com/documents/10.0.0/api/twisted.internet.interfaces.IReactorUDP.listenUDP.html

"Returns object which provides IListeningPort."

However:
```
>>> from twisted.internet import reactor
>>> from twisted.internet.protocol import DatagramProtocol
>>> from twisted.internet.interfaces import IListeningPort
>>> port = reactor.listenUDP(9123, DatagramProtocol())
>>> IListeningPort.providedBy(port)
False
```

The fix seems simple enough, add interfaces.IListeningPort to the list of interfaces in t.i.udp.Port

I noticed this bug because http://twistedmatrix.com/documents/10.0.0/api/twisted.internet.interfaces.IListeningPort.html says:

"Known implementations: twisted.internet.iocpreactor.tcp.Port, twisted.internet.tcp.Port"

Which I assume is auto-generated.

<details><summary>Searchable metadata</summary>

```
trac-id__4462 4462
type__defect defect
reporter__ghazel ghazel
priority__normal normal
milestone__ 
branch__branches_udp_listening_port_4462 branches/udp-listening-port-4462
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1275093307000000 1275093307000000
changetime__1277524714000000 1277524714000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29364]) Branching to 'udp-listening-port-4462'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/udp-listening-port-4462)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess set owner to @exarkun|
|-|-|

Thanks for working on this, exarkun, and thanks for the bonus unit tests!

 * in test_unix.py, "L{IReactorTCP.connectTCP} returns an object providing L{IConnector}." ==> "L{IReactorUNIX.connectUNIX} returns an object providing L{IConnector}."
 * test_tcp.py needs a copyright bump

Other than that, I think this is good to merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [29375]) Merge udp-listening-port-4462

Author: exarkun
Reviewer: jesstess
Fixes: [#4462](https://github.com/twisted/twisted/issues/4462)

Declare that the UDP Port classes implement IListeningPort, which they do.
Also add a couple similar tests for TCP connectors implementing IConnector.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
